core/cat/mad_hatter/mad_hatter.py

```diff
@@ -25,6 +26,43 @@ def __init__(self, ccat):
         self.ccat = ccat
         self.hooks, self.tools, self.plugins = self.find_plugins()
 
+    def install_plugin(self, zip_plugin):
+        plugin_folder = self.ccat.get_plugin_path()
+        zip_obj = Zip(zip_plugin)
+        zip_obj.unzip(plugin_folder)
+        zip_name = zip_obj.get_name()
+
+        plugin_name = zip_name.replace(".zip", "")
+        plugin_folder = path.join(plugin_folder, plugin_name)
+        plugin, tool = self.find_plugin(plugin_folder)
+        if plugin:
+            self.plugins.append(plugin)
+        if tool:
+            self.tools += tool
+
+        self.ccat.bootstrap()
```

Comment:
if we call bootstrap here, there is no need to call `find_plugin` above, because the `mad_hatter` makes the discovery at instantiation